flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableFactory.java

```diff
@@ -112,4 +141,75 @@ public HiveTableFactory(HiveConf hiveConf) {
 		return new HiveTableSink(new JobConf(hiveConf), tablePath, table);
 	}
 
+	@Override
+	public FunctionDefinition createFunctionDefinition(String name, CatalogFunction catalogFunction) {
```

Comment:
To have consistent APIs, it might be good to use ObjectPath in place of name.

Reply:
AFAICT, `createFunctionDefinition()` is a bit different from `createTableSource/Sink()`. Its implementation should be only about how to initialize a func definition from the CatalogFunction. The name is only for building the func definition rather than accessing something from catalog APIs, thus we don't need the database name.

Reply:
Actually, a factory does not need to know the name. `ObjectPath` in `createTableSource/Sink()` is only a temporary solution, right? If we want to pass it for debugging purposes we should pass the entire `ObjectIdentifier`. But I would prefer a clean signature `createXXX(CatalogXXX)`.

Reply:
@twalthr `HiveTableFactory#createTableSource/Sink()` does need the ObjectPath, not for debugging purpose, but to access partition information via Catalog APIs.